torchbenchmark/models/hf_T5/__init__.py

```diff
@@ -0,0 +1,56 @@
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import NLP
+from transformers import T5Tokenizer, T5ForConditionalGeneration
+from datasets import load_dataset
+
+class Model(BenchmarkModel):
+    task = NLP.LANGUAGE_MODELING
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+
+        batch_size = 8
+        seq_length = 512
+
+        input_ids = torch.randint(0, 5000, (batch_size, seq_length)).to(device)
+        decoder_ids = torch.randint(0, 5000, (batch_size, seq_length)).to(device)
+
+        self.model = T5ForConditionalGeneration.from_pretrained('t5-small').to(device)
+        self.train_inputs = {'input_ids': input_ids, 'labels': decoder_ids}
+        self.eval_inputs = {k: v[0:1] for k,v in self.train_inputs.items()}
+
+    def get_module(self):
+        if self.jit:
+            raise NotImplementedError()
+        return self.model, self.eval_inputs
+
+    def train(self, niter=3):
+        if self.jit:
+            raise NotImplementedError()
+        optimizer = optim.Adam(self.model.parameters())
```

Comment:
this is beyond the scope of this PR, but @wconstab we should consider outlining constructors in `pre_train` or something. I'm not sure if we really want to measuring how fast we can construct an optimizer object.

Reply:
that's a good point - I'll move it to the init.